### PR TITLE
Minor changes / additions

### PR DESCRIPTION
**Pull request by andrew-k9**
# Minor Quality Improvements

## Description

* added additional functionality to the Page class
  * added more to .gitignore
  * added a PR template
  * added input data from legacy system
  * added a 'ruby-modules' folder for any non-testing related classes/ methods we'll make
  * cleaning up some whitespace, spelling, etc

## Status

No more commits will be added to this unless CR requires it

## Functionality 

Basically just preparing for merging the legacy system into this. No added functionality aside from the helper methods defined

## Deployment

Merge after CR is accepted

### CODE REVIEW

**Comment by andrew-k9** [comment]
Also, if you could open a separate PR to comment the other methods/ classes you wrote that would be great! I don't want us to add something and then have it be confusing to anyone from the outside looking in 👀

File: pages/page.rb [hunk 1]
```diff
@@ -20,19 +25,103 @@ def navigate_to url
...
+    # 
+    #   get_input_json_data:       Gets input for a specific page from the 'data' folder. Any call to this method
+    #                              **MUST** specify which sub-folder to use
+    # 
+    #   inputs:        filename        - Filename + sub-folder of the data we need. ie 'appointment-info/estimate.json'
+    #
+    #   returns:                         hash of the json file 
+    #
+    #   throws:        ArgumentError   - when file not found
+    def get_input_json_data filename
```
Comment (patmccler) [inline pages/page.rb:100]: Does this belong on the Page object? It seems more related to setup and getting everything up and running.
Stuff we'd want to do before the page is even looked at.

All the other methods here are related to how the page works

Maybe we should create a module for input-type helper methods

File: pages/page.rb [hunk 2]
```diff
@@ -20,19 +25,103 @@ def navigate_to url
...
-# not sure about that, need to learn selenium still
+    # 
+    #   click_button_at_position:     clicks a button at the nth element of an array of buttons
+    # 
+    #   inputs:        button_identifier        - Selenium element set to the scroll wheel
+    #                  index                    - index of the HTML element to select
+    #
+    #   throws:        ArgumentError  - when the element given is **not** a selector
+    def click_button_at_position button_identifier, index
+        button = @wait.until{ @driver.find_elements(button_identifier)[index] }
```
Comment (patmccler) [inline pages/page.rb:45]: is "button Identifier" similar to the element_identifier from other methods?
Is it supposed to be {:how => "what"}  like the rest?

What does "Selenium element set to the scroll wheel" have to do with an array of button elements?

File: pages/page.rb (hunk 1 above)
Comment (patmccler) [reply pages/page.rb:100]: @andrew-k9  Let's put this in as is for now, and move it if/when we  refactor the add home workflow to take the input the same as other workflows (with the command)

File: pages/page.rb (hunk 2 above)
Comment (andrew-k9) [reply pages/page.rb:45]: 1. Yup, I was going off of the click_button from above (I may have changed it but I forgot)

2. I copy my comments and change the body. Forgot to update it there

Comment (patmccler) [reply pages/page.rb:45]: Cool, I thought so but wanted to make sure. I'll update both when I go through and comment the methods that are already there (sorry, I'll do that soon!)

**Review by patmccler — APPROVED** [review]
Looks good! After discussion lets ship it
